### PR TITLE
parser: add sql comments support

### LINKED ISSUE #1: parser: support SQL comments
Extend the parser so that he supports comments as specified here (https://www.sqlite.org/lang_comment.html).

### PR DESCRIPTION
**Pull request by Shivam010**
Closes #1

### CODE REVIEW

File: internal/parser/parser_test.go [hunk 1]
```diff
@@ -9812,6 +9812,97 @@ func TestSingleStatementParse(t *testing.T) {
...
+			"COMMIT -- making a commit",
+			&ast.SQLStmt{
+				CommitStmt: &ast.CommitStmt{
+					Commit: token.New(1, 1, 0, 6, token.KeywordCommit, "COMMIT"),
+				},
+			},
+		},
+		{
+			`Commented FROM clause in SELECT`,
+			"SELECT /** FROM*/ users",
```
Comment (tsatke) [inline internal/parser/parser_test.go:9826]: You are not testing from `/*` to end of input, or did I miss it?

**Review by tsatke — CHANGES_REQUESTED** [review]
Test from `/*` until end of input is missing

`SELECT * FROM myTable /* WHERE myCol > 4`

File: internal/parser/parser_test.go (hunk 1 above)
Comment (Shivam010) [reply internal/parser/parser_test.go:9826]: Trailing comment line no. [9888](https://github.com/xqueries/xdb/pull/46/files#diff-58edabd268c2fcdc59e6065c182a8296R9888) `"SELECT 2 /* trailing comment"`

Comment (tsatke) [reply internal/parser/parser_test.go:9826]: missed it

**Review by tsatke — APPROVED** [review]

**Pull request merged**